app/src/main/java/com/xa/xpensauditor/MonthWiseVis.java

```diff
@@ -140,30 +143,30 @@
                             dumm = snapshot.child("Day").getValue().toString();
                             dumm1 = Integer.valueOf(dumm);
                             System.out.println("dummmmm   " + dumm1 + "    " + dayyy);
-                            if (dayyy == dumm1) {
+                            Date date = new GregorianCalendar(y2, ii-1, dayyy).getTime();
+
+                            if(sortedMap.containsKey(date)){
                                 System.out.println("inside");
-                                int temp = 0;
-                                temp = hm.get(dayyy);
-                                temp = temp + am1;
-                                System.out.println("dummmmm1   " + dayyy + "    " + temp);
-                                hm.put(dayyy, temp);
+                                int temp=0;
+                                temp=sortedMap.get(date);
+                                temp=temp+am1;
+                                System.out.println("dummmmm1   "+dayyy+"    "+temp);
+                                sortedMap.put(date,temp);
+                                System.out.println("Dates printing-4   ");
                             } else {
                                 dayy = snapshot.child("Day").getValue().toString();
                                 dayyy = Integer.parseInt(dayy);
-                                hm.put(dayyy, am1);
+                                sortedMap.put(date,am1);
                             }
                         }
 
-                        ArrayList<Integer> sortedKeys
-                                = new ArrayList<Integer>(hm.keySet());
-
-                        Collections.sort(sortedKeys);
-
-                        // Display the TreeMap which is naturally sorted
-                        for (Integer x : sortedKeys) {
-                            data.add(new ValueDataEntry(x, hm.get(x)));
+                        for (Map.Entry< Date,Integer> entry :
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.